rasterio/_warp.pyx

```diff
@@ -964,13 +966,15 @@ cdef class WarpedVRTReaderBase(DatasetReaderBase):
 
         cdef GDALRasterBandH hBand = NULL
         src_alpha_band = 0
+        dst_alpha_band = 0
```

Comment:
```suggestion
        cdef int dst_alpha_band = 0
```

Reply:
@snowman2 @vincentsarago I think we're okay in 1.3.x without the explicit cdef. We're not accessing this value often, so the performance benefits of cdef'ing are small. Let's hold off on a complete refactor of the function.

Reply:
I agree that adding cdef in this PR may be out of scope. But, I think it is something worth looking into in the future.